Add gamepad arm controls

connorreinh opened the pull request:
Added gamepad controls for the arm as described in this image: 
![image](https://user-images.githubusercontent.com/43483846/126055270-3c99333f-23c5-40d4-afc6-4a077e5688c3.png)

I'll need someone else to test this since I don't have access to the physical rover. Some specific things to test for include making sure that the directions of motor movement actually match what's in the picture and that I didn't get the signs flipped.

To use the controls, simply plug two USB controllers into the computer running mission control. PS4, Xbox 360, Xbox One, and probably other controllers work.

Commit:
Add gamepad arm controls

evstraw reviewed (approved):
Sorry for the delay on this. Code looks good to me. @nhatch @Jacob6819 do you think you'd be able to test the controls in the lab on Friday (or some other day), if you get time?

nhatch commented:
I'm gonna copy the feedback I put in the Discord channel:

A small problem emerged in on-rover testing. It looks like the gamepad drive commands are being sent to the rover at a very high frequency (like 30 hz or something?) and I think it is overwhelming some of the motor boards / overflowing the CAN queue. (The symptom is that sometimes the wheels keep moving for one second after we release the control stick, which usually means that the set-velocity-zero command was dropped from the CAN queue.) Do you think you could throttle the gamepad controls so it updates at the same UPDATE_PERIOD_MILLIS as the keyboard controls?

As discussed on the Discord, I think we can also decrease the UPDATE_PERIOD_MILLIS to 200 ms, down from 700 ms.

Ideally we should make it so the command is still sent immediately if the previous or new velocity was/is zero, to avoid perceptible delays when starting or stopping the rover. But otherwise we could just rely on the update() method (in rover_commands.ts).

nhatch reviewed (changes requested):
I'm gonna copy the feedback I put in the Discord channel:

A small problem emerged in on-rover testing. It looks like the gamepad drive commands are being sent to the rover at a very high frequency (like 30 hz or something?) and I think it is overwhelming some of the motor boards / overflowing the CAN queue. (The symptom is that sometimes the wheels keep moving for one second after we release the control stick, which usually means that the set-velocity-zero command was dropped from the CAN queue.) Do you think you could throttle the gamepad controls so it updates at the same UPDATE_PERIOD_MILLIS as the keyboard controls?

As discussed on the Discord, I think we can also decrease the UPDATE_PERIOD_MILLIS to 200 ms, down from 700 ms.

Ideally we should make it so the command is still sent immediately if the previous or new velocity was/is zero, to avoid perceptible delays when starting or stopping the rover. But otherwise we could just rely on the update() method (in rover_commands.ts).

Commit:
Use 'export default' where appropriate

Commit:
Rename ControllerComponent to GamepadComponent

Commit:
Prevent overflowing of CAN queue

Made RoverCommands only send commands immediately when the difference between a motor's current power and requested power is greater than some epsilon value (currently 0.05).

Commit:
Improve RoverCommands packet sending

Nanolacker commented:
Sorry for the very long delay @nhatch . I made the changes you requested. Could you try using a gamepad on the rover to see if everything works correctly?
